packages/react-native/Libraries/StyleSheet/processBackgroundImage.js

```diff
@@ -0,0 +1,208 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ * @flow
+ */
+
+'use strict';
+
+import type {____ColorValue_Internal} from './StyleSheetTypes';
+
+const processColor = require('./processColor').default;
+
+const LINEAR_GRADIENT_REGEX = /linear-gradient\s*\(((?:\([^)]*\)|[^())])*)\)/g;
+const COLOR_STOP_REGEX =
+  /\s*((?:(?:rgba?|hsla?)\s*\([^)]+\))|#[0-9a-fA-F]+|[a-zA-Z]+)(?:\s+([0-9.]+%?))?\s*/g;
+const DIRECTION_REGEX =
+  /^to\s+(?:top|bottom|left|right)(?:\s+(?:top|bottom|left|right))?/;
+const ANGLE_UNIT_REGEX = /^.*?(deg|grad|rad|turn)$/;
+const TRANSPARENT = 0; // rgba(0, 0, 0, 0)
+const TO_BOTTOM_START_END_POINTS = {
+  start: {x: 0.5, y: 0},
+  end: {x: 0.5, y: 1},
+};
+
+export type BackgroundImagePrimitive = {
+  type: 'linearGradient',
+  start: {x: number, y: number},
+  end: {x: number, y: number},
+  colorStops: $ReadOnlyArray<{
+    color: ____ColorValue_Internal,
+    position: number,
+  }>,
+};
```

Comment:
We probably want to keep this in `StyleSheetTypes`, since apart from the processor, this forms the allowed style input.

A nit here, is that `Primitive` started propagating after "filter primitive" was used, referring to something in the filters spec unrelated to the actual input. I would love to avoid it in the future if we can.

The spec tends to refer to a single usage of this in a background to be either "background image" or "background image layer" to refer to what this is representing. And each background layer itself an `<image>`, which is either a either `<url>` or `<gradient>`, where then a `<gradient>` can be one of multiple types. But these may be used outside backgrounds.

Putting that altogether, I think this might make sense as something like a `GradientValue` type, so we can built that hierarchy in the future, and reuse this for other places with gradients.